examples/Conversational_agents_using_tools.ipynb

```diff
@@ -0,0 +1,1095 @@
+      "Final Answer: As of April 17, 2023, the population of Canada is 38,662,830.\u001B[0m\n",
+      "\n",
+      "\u001B[1m> Finished chain.\u001B[0m\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "'As of April 17, 2023, the population of Canada is 38,662,830.'"
+      ]
+     },
+     "execution_count": 16,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "agent_executor.run(\"How many people live in canada as of 2023?\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "9d28d73f",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "\n",
+      "\u001B[1m> Entering new AgentExecutor chain...\u001B[0m\n",
+      "\u001B[32;1m\u001B[1;3mThought: I need to search for the current population of Mexico.\n",
+      "Action: Search\n",
+      "Action Input: \"current population of Mexico\"\u001B[0m\n",
+      "\n",
+      "Observation:\u001B[36;1m\u001B[1;3mMexico, officially the United Mexican States, is a country in the southern portion of North America. It is bordered to the north by the United States; to the south and west by the Pacific Ocean; to the southeast by Guatemala, Belize, and the Caribbean Sea; and to the east by the Gulf of Mexico.\u001B[0m\u001B[32;1m\u001B[1;3mThat's not the answer to the question, I need to refine my search.\n",
+      "Action: Search\n",
+      "Action Input: \"population of Mexico 2023\"\u001B[0m\n",
+      "\n",
+      "Observation:\u001B[36;1m\u001B[1;3m128,455,567\u001B[0m\u001B[32;1m\u001B[1;3mI now know the final answer.\n",
+      "Final Answer: As of 2023, the population of Mexico is 128,455,567.\u001B[0m\n",
+      "\n",
+      "\u001B[1m> Finished chain.\u001B[0m\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "'As of 2023, the population of Mexico is 128,455,567.'"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "agent_executor.run(\"how about in mexico?\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "62356eed",
+   "metadata": {},
+   "source": [
+    "## Knowledge base\n",
+    "\n",
+    "Create a custom vectorstore for the Agent to use as a Tool to answer questions with. We'll store the results in [Pinecone](https://docs.pinecone.io/docs/quickstart), which is supported by LangChain ([Docs](https://python.langchain.com/en/latest/modules/indexes/vectorstores/examples/pinecone.html), [API reference](https://python.langchain.com/en/latest/reference/modules/vectorstore.html)). For help getting started with Pinecone or other vector databases, we have a [cookbook](https://github.com/openai/openai-cookbook/blob/colin/examples/vector_databases/Using_vector_databases_for_embeddings_search.ipynb) to help you get started.\n",
```

Comment:
I see "tool" is lower case in some places and upper case in others (like here). Recommend using lowercase everywhere.

Reply:
Agreed, well-spotted - the only places it remains capitalised are at the start of sentences or as a Class name